clients/go/pkg/worker/jobPoller_test.go

```diff
@@ -51,6 +52,7 @@ func (suite *JobPollerSuite) BeforeTest(_, _ string) {
 		closeSignal:    make(chan struct{}),
 		remaining:      0,
 		threshold:      int(math.Round(float64(DefaultJobWorkerMaxJobActive) * DefaultJobWorkerPollThreshold)),
+		shouldRetry:    func(_ context.Context, _ error) bool { return false },
```

Comment:
Should we introduce a type as we did for credentials provider? Considering this is then passed in a public function, it would help document its intent.

Reply:
I can change it so it's consistent with the other usage but I'm not sure. It's true that it gives you a small description of what it is but the parameter name does the same. Also, it covers the function signature which you need to implement. Besides, the `retryPredicate` in the commands package isn't exported so you can't see what it is and how it's documented. Wdyt?

Reply:
I think it might be ambiguous for users - not exactly what it does, but what should they pass? What should they implement here?

Reply:
I can change both interfaces to receive a CredentialsProvider (like you said in another comment) since that's an exported and documented interface. It's a breaking change but users are probably not using the New....Command methods directly in the command package

Reply:
Internally I guess we can use the function directly - I guess using CredentialsProvider would only be for the "public" stuff anyway. That way it's a little less coupled, but it's maybe a bit clearer to the users what should be passed? I guess the only concern here is that we need to be very specific that it should be the exact same used to configure the pb.GatewayClient, since if they pass another one we wouldn't use the "auth" part of it (does that make sense?).

wdyt of this idea? I feel like this is a better approach from a user pov - though like you said, users probably aren't creating the worker directly :thinking: 

If you like the idea then go for it, but if it sounds weird to you then I won't insist, as I'm only 80% sure about it :wink:

Reply:
Makes sense. It's not ideal but I think is as good as it gets if we want to keep exposing this builder. Imo the underlying issue here is that our API is way too big and exposes unnecessary things

Reply:
Makes sense. It's not ideal but I think is as good as it gets if we want to keep exposing this builder. Imo the underlying issue here is that our API is way too big and exposes unnecessary things